wp-includes/rest-api/endpoints/class-wp-rest-customize-controls-controller.php

```diff
@@ -0,0 +1,348 @@
+	 */
+	public function ensure_customize_manager() {
+		global $wp_customize;
+		if ( empty( $wp_customize ) ) {
+			$wp_customize = new WP_Customize_Manager(); // WPCS: global override ok.
+		}
+		if ( ! did_action( 'customize_register' ) ) {
+
+			/** This action is documented in wp-includes/class-wp-customize-manager.php */
+			do_action( 'customize_register', $wp_customize );
+		}
+		return $wp_customize;
+	}
+
+	/**
+	 * Registers the routes for the objects of the controller.
+	 *
+	 * @since ?.?.?
+	 * @access public
+	 *
+	 * @see register_rest_route()
+	 */
+	public function register_routes() {
+
+		register_rest_route( $this->namespace, '/' . $this->rest_base, array(
+			array(
+				'methods'             => WP_REST_Server::READABLE,
+				'callback'            => array( $this, 'get_items' ),
+				'permission_callback' => array( $this, 'get_items_permissions_check' ),
+				'args'                => $this->get_collection_params(),
+			),
+			'schema' => array( $this, 'get_public_item_schema' ),
+		) );
+
+		register_rest_route( $this->namespace, '/' . $this->rest_base . '/(?P<control>[\w-]+)', array(
+			'args' => array(
+				'control' => array(
+					'description' => __( 'An alphanumeric identifier for the control.' ),
+					'type'        => 'string',
+				),
+			),
+			array(
+				'methods'             => WP_REST_Server::READABLE,
+				'callback'            => array( $this, 'get_item' ),
+				'permission_callback' => array( $this, 'get_item_permissions_check' ),
+				'args'                => array(
+					'context' => $this->get_context_param( array(
+						'default' => 'view',
+					) ),
+				),
+			),
+			'schema' => array( $this, 'get_public_item_schema' ),
+		) );
+	}
+
+	/**
+	 * Retrieves the control's schema, conforming to JSON Schema.
+	 *
+	 * @since 4.?.?
+	 * @access public
+	 *
+	 * @return array Item schema data.
+	 */
+	public function get_item_schema() {
+
+		$schema = array(
+			'$schema'    => 'http://json-schema.org/schema#',
+			'title'      => 'control',
+			'type'       => 'object',
+			'properties' => array(
+				'allow_addition'  => array(
+					'description' => __( 'If to show UI for adding new content' ),
+					'type'        => 'boolean',
+					'context'     => array( 'embed', 'view' ),
+					'readonly'    => true,
+				),
+				'choices'         => array(
+					'description' => __( 'List of choices for radio/select.' ),
+					'type'        => 'object',
+					'context'     => array( 'embed', 'view', 'edit' ),
+					'readonly'    => true,
+				),
+				'description'     => array(
+					'description' => __( 'Control description.' ),
+					'type'        => 'string',
+					'context'     => array( 'embed', 'view', 'edit' ),
+					'readonly'    => true,
+				),
+				'id'              => array(
+					'description' => __( 'Identifier for the control.' ),
+					'type'        => 'string',
+					'context'     => array( 'embed', 'view' ),
+					'readonly'    => true,
+				),
+				'instance_number' => array(
```

Comment:
@miina The purpose of the `instance_number` was to ensure that if you have registered multiple controls with the same priority, that the controls end up getting listed in the same order as they were registered in, and also to ensure the sorting is stable. I'm not sure if we implemented this to mask a more fundamental problem of core/plugins/themes not giving separate priorities. I think it would be better for stable sorting to fallback to comparing the section titles if they end up having the same priorities. So yeah, I think `instance_number` can be omitted from the REST API response.